lua/cmake/utils.lua

```diff
@@ -89,7 +90,18 @@ function utils.get_current_target(parameters)
     return nil, nil
   end
 
-  local target_dir = vim.fn.fnamemodify(target, ':h')
+  local target_dir
+  if parameters['runDir'] ~= nil then
+    local run_path = Path:new(parameters['runDir'])
+    if run_path:is_absolute() then
+      target_dir = run_path:normalize(vim.loop.cwd())
+    else
+      target_dir = Path:new(build_dir .. parameters['runDir']):normalize(vim.loop.cwd())
+    end
+    target = Path:new(target):make_relative(target_dir)
+  else
+    target_dir = vim.fn.fnamemodify(target, ':h')
```

Comment:
The Plenary:parent() doesn't seem to work as expected. This way seems to return the correct value for `target` in the easiest way, since it takes the full path of `target`, and converts it into a relative path with respect to `target_dir`. This then passes `target_dir` to `async_run` as the `cwd` and the relative path to the target as the executable to run.